app/controllers/authors_controller.rb

```diff
@@ -24,7 +21,11 @@ def create
     end
   end
 
-  # PATCH/PUT /authors/1
+  def issue_to_author
+    GithubIssueWorker.perform_async(github_payload)
```

Comment:
1- the secret from the webhook is the solution i found, will push the updated code soon.
2- mapping with factory is what i'm thinking about, each action can be separated into a different class that can handle it's own business logic, and you would be having a hash map which would lead you to correct class with the help of factory design pattern, is what i would do.

Reply:
thx 👍